python_modules/dagster/dagster/components/core/defs_module.py

```diff
@@ -240,7 +273,4 @@ def load_yaml_component(context: ComponentLoadContext) -> Component:
         components.append(obj.load(attributes, context))
 
     check.invariant(len(components) > 0, "No components found in YAML file")
-    if len(components) == 1:
-        return components[0]
-    else:
-        return CompositeYamlComponent(components)
+    return CompositeYamlComponent(components)
```

Comment:
This moves us to a world where the [component loaded from a YAML folder is always a wrapper again](https://github.com/dagster-io/dagster/blob/release-1.10.8/python_modules/dagster/dagster/components/core/defs_module.py#L204), which does make some of the below tests a bit less clean